packages/resolve/src/request-resolver.ts

```diff
@@ -117,6 +123,35 @@ export function createRequestResolver(options: IRequestResolverOptions): Request
     }
   }
 
+  /**
+   *
+   * @param request - the original request
+   * The function generates all the possible aliased requests, and falls back to the original request if all else failed
+   */
+  function* aliasRequestPaths(request: string) {
+    for (const [aliasedFrom, aliasedTo] of Object.entries(aliases)) {
+      if (aliasedFrom.endsWith('$') && request === aliasedFrom.slice(0, aliasedFrom.length - 1)) {
+        if (aliasedTo === false) {
+          yield false;
+        }
+        yield aliasedTo;
+      } else if (!request.endsWith('$') && request.startsWith(aliasedFrom)) {
+        if (aliasedTo === false) {
+          yield false;
+        } else if (/\.\w+$/.test(aliasedTo)) {
```

Comment:
@AviVahl  what do you think about bringing in path as a dependency for resolve so I can use path.extname and path.join?